src/clients/c++/experimental_api_v2/library/grpc_client.cc

```diff
@@ -249,10 +281,53 @@ InferenceServerGrpcClient::GetModelConfig(
   return err;
 }
 
+Error
+InferenceServerGrpcClient::Infer(
+    std::shared_ptr<InferResultGrpc>* result, const InferOptions& options,
```

Comment:
Our use of unique_ptr and shared_ptr in the interfaces isn't really needed is it? Are we enforcing some allocation policy that we don't need to? Do we keep copies of result, input or output? If not perhaps we should pass by const& (or const* in the vectors), and return via pointer. The caller can then wrap the raw pointers in shared/unique if they want. What do you think?

Reply:
You are right. We must not enforce any policy in our API level. Fixed,